app/Repositories/ProfileRepository.php

```diff
@@ -9,44 +10,50 @@
 class ProfileRepository
 {
     protected $model;
+
     public function __construct(Profile $model)
     {
         $this->model = $model;
     }
 
-    public function getProfile($request, $address) {
-        return $this->model->with(['users.circle.protocol','users.teammates','users.histories.epoch'])->byAddress($address)->first();
+    public function getProfile($address, $relationships = [])
+    {
+        return $this->model->with($relationships)->byAddress($address)->first();
     }
 
-    public function saveProfile($request) {
-        $data = $request->only('skills','bio','telegram_username',
-            'discord_username','twitter_username','github_username','medium_username','website');
+    public function saveProfile($request)
+    {
+        $data = $request->only('skills', 'bio', 'telegram_username',
+            'discord_username', 'twitter_username', 'github_username', 'medium_username', 'website');
         $profile = $request->profile;
         $profile->update($data);
-        $profile->load(['users.circle.protocol','users.teammates','users.histories.epoch']);
+        $profile->load(['users.circle.protocol', 'users.teammates', 'users.histories.epoch']);
         return $profile;
     }
 
-    public function uploadProfileAvatar($request, $address) {
-        $profile =  $this->model->byAddress($address)->first();
-        if($profile) {
+    public function uploadProfileAvatar($request, $address)
+    {
+        $profile = $this->model->byAddress($address)->first();
+        if ($profile) {
             $file = $request->file('file');
             $image = Image::make($file);
             $height = $image->height();
             $width = $image->width();
 
-            if($width> 240) {
-                $height = $height * 240/$height;
-                $width = $width * 240/$width;
+            if ($width > 240) {
+                $height = $height * 240 / $height;
+                $width = $width * 240 / $width;
             }
 
             $resized = $image
-                ->resize($width, $height, function ($constraint) { $constraint->aspectRatio(); } )
-                ->encode($file->getCLientOriginalExtension(),80);
-            $new_file_name = Str::slug(pathinfo(basename($file->getClientOriginalName()), PATHINFO_FILENAME)).'_'.time().'.'.$file->getCLientOriginalExtension();
+                ->resize($width, $height, function ($constraint) {
```

Comment:
can you move this into its own PR please, so we can ship it sooner than the access control stuff? i also assigned the Linear task for this to you and marked it as in progress

Reply:
Haven't started on that task yet (could work on that first if its of more priority), this is probably because of the IDE doing its formatting as i've made some changes to this file

Reply:
Ah, oops, I saw aspectRatio and jumped to conclusions